### ISSUE TITLE
CGroups Leak - Docker 1.12.6 and 17.03.2 

### ISSUE THREAD

**Issue by swade1987** [moby/moby#35601]
I am running the following Kubernetes cluster setup on Digital Ocean using [KET 1.6.3](https://github.com/apprenda/kismatic/releases/tag/v1.6.3)  
 
```
OS: CentOS 7
Kernel Version: 3.10.0-693.5.2.el7.x86_64
Docker with directlvm enabled
Swap disabled on all nodes
```
 
Note: I tested the below running both Docker 1.12.6 and 17.03.2 and the same issue happened.
 
I ran the daemonset which causes pods to get OOM killed and left it running for 33 minutes
 
I went onto one of the nodes and performed a `docker ps –a` and located the CgroupParent for the docker container.
 
I then browsed to that directory (/sys/fs/cgroup/memory/kubepods/burstable/pod6e5b24a8-c8b6-11e7-b5e8-a6275109595d) and saw the following:
 
```
drwxr-xr-x. 2 root root 0 Nov 13 21:05 447a0f0b4b1aad5d9ee27eae473451287c49f7634545f6382126d50bb824ea57
drwxr-xr-x. 5 root root 0 Nov 13 21:27 .
drwxr-xr-x. 2 root root 0 Nov 13 21:32 da7a4d442f03f51c30597ea52c74a418c8cd85ffe202e746908d32a525aa5a27
drwxr-xr-x. 2 root root 0 Nov 13 21:37 60bba159f801c40937166bbe9d2f541b9c904dcc15340a6ee68def268f78193d
```
 
The longer I left it, the more cgroups get leaked, this eventually causes the node to become NotReady and requires a reboot of the node.
 
The only fixes thus far are downgrading to 1.11.2 or use the official RHEL package for Docker 1.12.6

**Comment by tianshapjq** [comment]
/cc 
we also found this phenomenon in our setup. When deploying a containerized kubelet, the MEM of this container in _docker stats_ of grew up slowly. When reaching the memory limit it would get an OOM kill or swap out quite a lot, both led the kubelet dead finally.
And we found that the MEM of _docker stats_ just came from the memory.usage_in_bytes in cgroup directly. How does memory.usage_in_bytes calculated?
